basedirs/db.go

```diff
@@ -212,8 +220,35 @@ func (b *BaseDirs) storeGIDBaseDirs(tx *bolt.Tx, gidBase map[uint32]dgut.DCSs) e
 	return nil
 }
 
-func keyName(id uint32, path string) []byte {
-	return []byte(strconv.FormatUint(uint64(id), 10) + bucketKeySeparator + path)
+func keyName(id uint32, path string, age summary.DirGUTAge) []byte {
+	idBs := idToByteSlice(id)
+
+	ageBs := ageToByteSlice(age)
+
+	length := sizeOfUint32 + sizeOfUint16 + len(path)
+	b := bytes.NewBuffer(make([]byte, 0, length))
+
+	b.Write(idBs)
+	b.Write([]byte{bucketKeySeparatorByte})
```

Comment:
Can the bucketKeySeparatorByte const be stored as a byte slice, to avoid create two of these each time?

Alternatively, you can change this to a bunch of appends:

```go
	b := make([]byte, 0, length)
	b = append(b, idBs)
	b = append(b, bucketKeySeparator...)
	b = append(b, path...)
	b = append(b, bucketKeySeparator...)
	b = append(b, agsBs)
```

Reply:
We benchmarked append vs buffer write, and append was faster, so went with this, despite also creating a global var for bucketKeySeparatorByteSlice to use in checkAgeOfKeyIsAll()